src/DataStructures/SliceIterator.cpp

```diff
@@ -37,11 +39,74 @@ void SliceIterator::reset() {
   slice_offset_ = 0;
 }
 
+template <size_t VolumeDim>
+std::pair<std::unique_ptr<std::pair<size_t, size_t>[], decltype(&free)>,
+          std::array<std::pair<gsl::span<std::pair<size_t, size_t>>,
+                               gsl::span<std::pair<size_t, size_t>>>,
+                     VolumeDim>>
+volume_and_slice_indices(const Index<VolumeDim>& extents) noexcept {
+  // Compute the slice indices once since this is (surprisingly) expensive
```

Comment:
This comment doesn't need to be in the implementation if the header documents this idea.